### ISSUE TITLE
Cannot install - rack-test version issue

### ISSUE THREAD

**Issue by davama** [ytti/oxidized-web#149]
Was trying a fresh install

```
gem install oxidized-web
ERROR:  Error installing oxidized-web:
	rack-test requires Ruby version >= 2.2.2.
```

Same result. Building from source:

```
gem build *.gemspec
  Successfully built RubyGem
  Name: oxidized-web
  Version: 0.9.3
  File: oxidized-web-0.9.3.gem

gem install  *.gem
Fetching: puma-3.11.0.gem (100%)
Building native extensions.  This could take a while...
Successfully installed puma-3.11.0
Fetching: rack-1.6.8.gem (100%)
Successfully installed rack-1.6.8
Fetching: tilt-2.0.8.gem (100%)
Successfully installed tilt-2.0.8
Fetching: rack-protection-1.5.3.gem (100%)
Successfully installed rack-protection-1.5.3
Fetching: sinatra-1.4.8.gem (100%)
Successfully installed sinatra-1.4.8
Fetching: backports-3.10.3.gem (100%)
Successfully installed backports-3.10.3
Fetching: rack-test-0.8.2.gem (100%)
ERROR:  Error installing oxidized-web-0.9.3.gem:
	rack-test requires Ruby version >= 2.2.2.
```

Centos7 
Kernel: 3.10.0-693.5.2.el7.x86_64
Release: 7.4.1708

What can i try? First time i see these results.
Already installed pkgs mentioned [here](https://github.com/ytti/oxidized#centos-oracle-linux-red-hat-linux)

Thanks!

-dave

**Comment by raisullip** [comment]
yeah same issue here 

looks like oxidized-web rack-test requires Ruby version >= 2.2.2. even when installing old versions

[root@default-centos-72 ~]# gem install oxidized-web -v 0.0.1
Fetching: rack-2.0.3.gem (100%)
ERROR: Error installing oxidized-web:
rack requires Ruby version >= 2.2.2.

**Comment by laf** [comment]
For now run `gem install rack-test -v 0.6.3` then install oxidized-web

**Comment by raisullip** [comment]
no dice, 

[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.6.3
Fetching: rack-2.0.3.gem (100%)
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.

**Comment by laf** [comment]
```bash
[root@web01 captures]# ruby -v
ruby 2.0.0p648 (2015-12-16) [x86_64-linux]
[root@web01 captures]# gem list rack-test

*** LOCAL GEMS ***

rack-test (0.6.3)
```
That's on CentOS 7 (7.3).

The output of your install is different for me:

```bash
[root@web01 captures]# gem install rack-test -v 0.6.3
Fetching: rack-test-0.6.3.gem (100%)
Successfully installed rack-test-0.6.3
Parsing documentation for rack-test-0.6.3
Installing ri documentation for rack-test-0.6.3
1 gem installed
```

**Comment by raisullip** [comment]
looks like 0.5.0 does it 

[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.6.3
Fetching: rack-2.0.3.gem (100%)
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.6.2
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.6.1
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.7
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.6
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.5
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.4
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.3
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.2
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.1
ERROR:  Error installing rack-test:
	rack requires Ruby version >= 2.2.2.
[vagrant@default-centos-72 ~]$ gem install rack-test -v 0.5.0
Fetching: rack-test-0.5.0.gem (100%)
Successfully installed rack-test-0.5.0
Parsing documentation for rack-test-0.5.0
Installing ri documentation for rack-test-0.5.0
1 gem installed

**Comment by raisullip** [comment]
any advice against using 0.5.0 ?

**Comment by laf** [comment]
None from me I'm afraid. It must be a dependency of something we rely on.

@ytti Any thoughts on this?

**Comment by davama** [comment]
@laf  Thank you! That did the trick
> For now run gem install rack-test -v 0.6.3 then install oxidized-web

```
gem install rack-test -v 0.6.3
Fetching: rack-test-0.6.3.gem (100%)
Successfully installed rack-test-0.6.3
Parsing documentation for rack-test-0.6.3
Installing ri documentation for rack-test-0.6.3
1 gem installed

gem build oxidized-web.gemspec 
  Successfully built RubyGem
  Name: oxidized-web
  Version: 0.9.3
  File: oxidized-web-0.9.3.gem

gem install oxidized-web-0.9.3.gem 
Fetching: multi_json-1.12.2.gem (100%)
Successfully installed multi_json-1.12.2
Fetching: sinatra-contrib-1.4.7.gem (100%)
Successfully installed sinatra-contrib-1.4.7
Fetching: haml-4.0.7.gem (100%)

HEADS UP! Haml 4.0 has many improvements, but also has changes that may break
your application:

* Support for Ruby 1.8.6 dropped
* Support for Rails 2 dropped
* Sass filter now always outputs <style> tags
* Data attributes are now hyphenated, not underscored
* html2haml utility moved to the html2haml gem
* Textile and Maruku filters moved to the haml-contrib gem

For more info see:

http://rubydoc.info/github/haml/haml/file/CHANGELOG.md

Successfully installed haml-4.0.7
Fetching: rb-fsevent-0.10.2.gem (100%)
Successfully installed rb-fsevent-0.10.2
Fetching: ffi-1.9.18.gem (100%)
Building native extensions.  This could take a while...
Successfully installed ffi-1.9.18
Fetching: rb-inotify-0.9.10.gem (100%)
Successfully installed rb-inotify-0.9.10
Fetching: sass-listen-4.0.0.gem (100%)
Successfully installed sass-listen-4.0.0
Fetching: sass-3.5.3.gem (100%)
Successfully installed sass-3.5.3
Fetching: emk-sinatra-url-for-0.2.1.gem (100%)
Successfully installed emk-sinatra-url-for-0.2.1
Fetching: htmlentities-4.3.4.gem (100%)
Successfully installed htmlentities-4.3.4
Successfully installed oxidized-web-0.9.3
Parsing documentation for multi_json-1.12.2
Installing ri documentation for multi_json-1.12.2
Parsing documentation for sinatra-contrib-1.4.7
Installing ri documentation for sinatra-contrib-1.4.7
Parsing documentation for haml-4.0.7
Installing ri documentation for haml-4.0.7
Parsing documentation for rb-fsevent-0.10.2
Installing ri documentation for rb-fsevent-0.10.2
Parsing documentation for ffi-1.9.18
Installing ri documentation for ffi-1.9.18
Parsing documentation for rb-inotify-0.9.10
Installing ri documentation for rb-inotify-0.9.10
Parsing documentation for sass-listen-4.0.0
Installing ri documentation for sass-listen-4.0.0
Parsing documentation for sass-3.5.3
Installing ri documentation for sass-3.5.3
Parsing documentation for emk-sinatra-url-for-0.2.1
Installing ri documentation for emk-sinatra-url-for-0.2.1
Parsing documentation for htmlentities-4.3.4
Installing ri documentation for htmlentities-4.3.4
Parsing documentation for oxidized-web-0.9.3
unable to convert "\x90" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/fonts/glyphicons-halflings-regular.eot, skipping
unable to convert "\xA1" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/fonts/glyphicons-halflings-regular.woff, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/diff_15x17.png, skipping
unable to convert "\x88" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/favicon.ico, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/oxidizing.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/oxidizing_40px.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/sort_asc.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/sort_asc_disabled.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/sort_both.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/sort_desc.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/sort_desc_disabled.png, skipping
unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/oxidized/web/public/images/versioning_18px.png, skipping
Installing ri documentation for oxidized-web-0.9.3
11 gems installed
```

Thanks again for the support!

**Comment by raisullip** [comment]
thanks for reopening it

**Comment by laf** [comment]
np, we need to find the proper fix.

**Comment by raisullip** [comment]
Agreed, let me know if I can help

**Comment by wk** [comment]
The hard dependency on ruby 2.2.2 for rack-test was introduced in rack-test 0.8.0, so versions up to and including 0.7.0 should work.

**Comment by wk** [comment]
This has been addressed in #165, uninstalling all versions of rack-test and building a fresh oxidized-web gem from master should pull the correct version for backwards compatibility down to ruby 2.0.
